inc/saiswitch.h

```diff
@@ -349,7 +349,14 @@ typedef enum _sai_switch_attr_t
     /** Nonvolatile storage required by both SAI and NPU in KB [sai_uint64_t]
      * Will be 0 for SAI_RESTART_TYPE_NONE */
     SAI_SWITCH_ATTR_NV_STORAGE_SIZE,
-
+    
+    /** Whether action list is mandatory for table creation [bool] 
+     * Default [false]*/
+    SAI_SWITCH_ATTR_ACL_TABLE_ACTION_LIST_MANDATORY,
+
+    /** List of actions supported [sai_u32_list_t] 
+     * List of all supported actions will be returned */
+    SAI_SWITCH_ATTR_ACL_TABLE_ACTION_LIST,
```

Comment:
Dell - Internal Use - Confidential
HI Atit,

Updated the comments in saiacl.h.

The SAI_SWITCH_ATTR_ACL_TABLE_ACTION_LIST_MANDATORY is a Boolean variable which will indicate whether action list is a mandatory attribute needed during table creation.

The SAI_SWITCH_ATTR_ACL_TABLE_ACTION_LIST will be the list of all the actions supported by the NPU defined in SAI. This action list can be queried and SAI_ACL_TABLE_ATTR_ACTION_LIST can be set.

If the SAI_SWITCH_ATTR_ACL_TABLE_ACTION_LIST_MANDATORY is true then application has to explicitly send the action list.



Thanks
Anbalagan

From: Atit Jain [mailto:notifications@github.com]
Sent: Wednesday, June 08, 2016 3:47 PM
To: opencomputeproject/SAI <SAI@noreply.github.com>
Cc: Natchimuth, Anbalagan <Anbalagan_Natchimuth@Dell.com>; Author <author@noreply.github.com>
Subject: Re: [opencomputeproject/SAI] Changes to add action list as part of table creation (#124)


In inc/saiswitch.h<https://github.com/opencomputeproject/SAI/pull/124#discussion_r66229553>:

> @@ -349,7 +349,14 @@ typedef enum _sai_switch_attr_t

>      /** Nonvolatile storage required by both SAI and NPU in KB [sai_uint64_t]

>       * Will be 0 for SAI_RESTART_TYPE_NONE */

>      SAI_SWITCH_ATTR_NV_STORAGE_SIZE,

> -

> +

> +    /** Whether action list is mandatory for table creation [bool]

> +     * Default [false]*/

> +    SAI_SWITCH_ATTR_ACL_TABLE_ACTION_LIST_MANDATORY,

> +

> +    /** List of actions supported [sai_u32_list_t]

> +     * List of all supported actions will be returned */

> +    SAI_SWITCH_ATTR_ACL_TABLE_ACTION_LIST,

What is use of SAI_SWITCH_ATTR_ACL_TABLE_ACTION_LIST_MANDATORY is this something that cant be knokced off while creating the tables. Also all ACL tables (in all stages) might not support same set of actions, so how will this global be useful. Do we assume mandatory to be same for all stages etc.
Typically Keys can be programmed to save on the TCAM width. What is the use of restricting actions, could ASICs get advantage on the width of action memory (say SRAM etc)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/opencomputeproject/SAI/pull/124/files/df7d7302506b3962664a4183acf3d37a390b7605#r66229553>, or mute the thread<https://github.com/notifications/unsubscribe/AMn3iMTsQDbcGmPmj18-had4WHdhazxbks5qJpaSgaJpZM4IwmuB>.